File: app/src/main/java/com/example/miniproject/ViewModel/SourceViewModel.java
package com.example.miniproject.ViewModel;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.example.miniproject.Database.Controller.DatabaseController;
import com.example.miniproject.Model.SourceModel;
import com.example.miniproject.Repository.SourceRepository;

import java.util.List;

public class SourceViewModel extends AndroidViewModel {
    private SourceRepository sourceRepository;

    public SourceViewModel(@NonNull Application application) {
        super(application);
        this.sourceRepository = new SourceRepository(application);
    }

    public void callApiAndSaveInDB(){
        sourceRepository.getSourceListFromWebAndInsertInDB();
    }

    public LiveData<List<SourceModel>> getSourceList() {
        return sourceRepository.getSourceListFromDb();
    }
}

File: app/src/main/java/com/example/miniproject/viewModel/SourceViewModel.kt
package com.example.miniproject.viewModel

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.example.miniproject.model.SourceModel
import com.example.miniproject.repository.SourceRepository
import javax.inject.Inject

class SourceViewModel @Inject constructor(private var sourceRepository: SourceRepository) : ViewModel() {

    fun callApiAndSaveInDB() {
        sourceRepository.sourceListFromWebAndInsertInDB
    }

    fun getSourceList(): LiveData<List<SourceModel>?>? {
        return sourceRepository.getSourceListFromDb()
    }
}